.github/workflows/benchmark-pull-request.yml

```diff
@@ -0,0 +1,145 @@
+            })
+      - name: Fail workflow for invalid comment
+        if: steps.check_comment.outputs.invalid == 'true'
+        run: |
+          echo "Invalid comment format detected. Failing the workflow."
+          exit 1
+      - id: get_approvers
+        run: |
+          echo "approvers=$(cat .github/CODEOWNERS | grep '^\*'  | tr -d '* ' | sed 's/@/,/g' | sed 's/,//1')" >> $GITHUB_OUTPUT
+      - uses: trstringer/manual-approval@v1
+        if: (!contains(steps.get_approvers.outputs.approvers, github.event.comment.user.login))
+        with:
+          secret: ${{ github.TOKEN }}
+          approvers: ${{ steps.get_approvers.outputs.approvers }}
+          minimum-approvals: 1
+          issue-title: 'Request to approve/deny benchmark for PR #${{ env.PR_NUMBER }}'
+          issue-body: "Please approve or deny the release for PR #${{ env.PR_NUMBER }}"
+          exclude-workflow-initiator-as-approver: false
+      - name: Get PR Details
+        id: get_pr
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const issue = context.payload.issue;
+            const prNumber = issue.number;
+            console.log(`Pull Request Number: ${prNumber}`);
+
+            const { data: pull_request } = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: prNumber,
+            });
+
+            return {
+              "headRepoFullName": pull_request.head.repo.full_name,
+              "headRef": pull_request.head.ref
+            };
+      - name: Set pr details env vars
+        run: |
+          echo '${{ steps.get_pr.outputs.result }}' | jq -r '.headRepoFullName'
+          echo '${{ steps.get_pr.outputs.result }}' | jq -r '.headRef'
+          headRepo=$(echo '${{ steps.get_pr.outputs.result }}' | jq -r '.headRepoFullName')
+          headRef=$(echo '${{ steps.get_pr.outputs.result }}' | jq -r '.headRef')
+          echo "prHeadRepo=$headRepo" >> $GITHUB_ENV
+          echo "prheadRef=$headRef" >> $GITHUB_ENV
+      - name: Print ENV
+        run: |
+          cat $GITHUB_ENV
+      - name: Checkout PR Repo
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ env.prHeadRepo }}
+          ref: ${{ env.prHeadRef }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 21
+      - name: Build and Assemble OpenSearch from PR
+        run: |
+          ./gradlew :distribution:archives:linux-tar:assemble -Dbuild.snapshot=false
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.UPLOAD_ARCHIVE_ARTIFACT_ROLE }}
+          role-session-name: publish-to-s3
+          aws-region: us-west-2
+      - name: Push to S3
```

Comment:
I believe we need a S3 cleanup step after `Trigger jenkins workflow to run gradle check`

Reply:
Do you mean to say delete the uploaded artifact after benchmark job has been submitted?

Reply:
Exactly, sorry about vague description

Reply:
I didn't add because it can take some time, upto 5-7 mins for the tarball to get downloaded for spinning up the cluster, if the jenkins agent nodes are not ready then even some more. 
However, I have added a lifecycle rule on the bucket to purge all the objects older than 30-days, I think I can reduce it even further to 1-day.

Reply:
>  didn't add because it can take some time, upto 5-7 mins after the job has started, for the tarball to get downloaded for spinning up the cluster,

Ah I see, may be we could design the benchmarks the same way we do Gradle check by polling Jenkins URL? In this case it will be very clear when job is done (and exact moment when cleanup / purge should take place)

Reply:
I thought of that, but some workloads, like `big5` can run for 5-6 hours, `vectorsearch` indexing runs for 10-hrs, I am afraid we might hit the timeout for runners. 
I can take an action item to have this clean up done in the jenkins job as part of `post` action in the related jenkinsfile. 
Let me know your thoughts.

Reply:
> I thought of that, but some workloads, like `big5` can run for 5-6 hours, `vectorsearch` indexing runs for 10-hrs, I am afraid we might hit the timeout for runners.

That is a good point

> I can take an action item to have this clean up done in the jenkins job as part of post action in the related jenkinsfile.
Let me know your thoughts.

I think N days purging ( lifecycle rule ) as you suggested before is the simplest way to go